File: LusoHealth/LusoHealthClient/LusoHealthClient.Server/Models/FeedbackAndReports/Report.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LusoHealthClient.Server.Models.FeedbackAndReports
{
	public class Report
	{
		[Key]
		public int Id { get; set; }

		public DateTime Timestamp { get; set; }

		[ForeignKey("Patient")]
		public string IdPatient { get; set; }

		[ForeignKey("Professional")]
		public string IdProfesional { get; set; }

		public string Description { get; set; }

		public ReportState State { get; set; }
	}
}

File: LusoHealth/LusoHealthClient/lusohealthclient.client/src/app/shared/models/profile/report.ts
export enum ReportState {
  Pending = 'Pending',
  Concluded = 'Concluded',
  Cancelled = 'Cancelled'
}

export class Report {
  id: number | null;
  timestamp: Date | null;
  idPatient: string | null;
  idProfesional: string | null;
  description: string | null;
  state: ReportState | null;

  constructor(
    id: number | null,
    timestamp: Date | null,
    idPatient: string | null,
    idProfesional: string | null,
    description: string | null,
    state: ReportState | null
  ) {
    this.id = id;
    this.timestamp = timestamp;
    this.idPatient = idPatient;
    this.idProfesional = idProfesional;
    this.description = description;
    this.state = state;
  }
}
